assets/js/data/cart/reducers.js

```diff
@@ -3,6 +3,33 @@
  */
 import { ACTION_TYPES as types } from './action-types';
 
+/**
+ * Sub-reducer for cart items array.
+ *
+ * @param   {Array}  state   cartData.items state slice.
+ * @param   {Object}  action  Action object.
+ */
+const cartItemsReducer = ( state = [], action ) => {
+	switch ( action.type ) {
+		case types.ITEM_QUANTITY_PENDING:
+			return state.map( ( cartItem ) => {
+				if ( cartItem.key === action.cartItemKey ) {
+					return {
+						...cartItem,
+						isQuantityPending: action.isQuantityPending,
+					};
+				}
+				return cartItem;
+			} );
```

Comment:
I wouldn't store `isQuantityPending` with the cart item because it means we're changing the shape of `cartItem` object and some will have this property, some will not.  Instead, I would add a new top_level property on the cart state `cartItemsQuantityPending` and it would simply be an array of cartItem keys.  The selector then just checks to see if the key is in the array.

So to be clear, the cart state would be:

```js
/**
  * @typedef CartState
  * @property {Object} cartData
  * @property {Object} metaData
  * @property {Array}  errors
  * @property {Array}  cartItemsQuantityPending
  */
```